tiledb/sm/rest/rest_client.cc

```diff
@@ -286,24 +281,21 @@ Status RestClient::post_query_submit(
       std::placeholders::_4,
       &scratch,
       query,
-      copy_state,
-      &user_buffers_overflowed);
+      copy_state);
 
   const Status st = curlc.post_data(
       url, serialization_type_, &serialized, std::move(write_cb));
 
-  // When 'user_buffers_overflowed' is true, we will return all of
-  // the known query data even if the response parsing was unsuccessful.
-  // This is a temporary work-around to allow returning an incomplete
-  // query when the client-side buffer is too small to completely
-  // store the returned query.
-  if (!user_buffers_overflowed) {
-    RETURN_NOT_OK(st);
-  }
-
   if (copy_state->empty()) {
-    return LOG_STATUS(Status::RestError(
-        "Error submitting query to REST; server returned no data."));
+    if (!st.ok()) {
+      return LOG_STATUS(Status::RestError(
+          "Error submitting query to REST; "
+          "server returned no data. "
+          "Curl error: " +
+          st.message()));
+    } else {
+      query->set_status(QueryStatus::INCOMPLETE);
+    }
```

Comment:
@tdenniston (cc: @Shelnutt2) I just changed this logic after a discussion with Seth. This ensures that we maintain the behavior where we can return an OK status with an incomplete query that has 0 data. This is how the client will determine that they did not allocate enough space for a single query.

So our **four** expected states are:
1. OK status with a complete query.
2. OK status with an incomplete query with non-zero buffer(s) size
3. OK status with an incomplete query with zero buffer(s) size
4. Non-OK status.

The client is expected to handle this scenarios as:
1. Do nothing.
2. Resubmit.
3. Increase buffer size and resubmit.
4. Examine the error message to determine the issue.

Reply:
Note that this depends on an in-progress REST-server patch to return an HTTP 200 for scenario 3. Currently, we are unable to path to scenario 3.

Reply:
Is it correct to unconditionally set the status to incomplete in this case?

Reply:
I think so: this means that the server responded with HTTP 200 but without any data.
- If any data was sent, our callback return value would trigger a curl error unless it was able to deserialize all data.
- If the connection was interrupted for any reason, the callback return value would trigger a curl error.

In other words: no curl error means that we did not receive any data, there wasn't a connection issue, and the server responded with a 200.

Reply:
Ok and does this still work if the query region of the array is actually empty? I.e. a complete query with 0 results.

Reply:
@tdenniston Yeah -- I edited my last comment while you were reading. I added the last sentence to address this: `If a query legitimately returns no data, we will still get a query object from the server, which will be deserialized into a COMPLETED status`

Reply:
Gotcha. It makes sense and good catch!

Reply:
I want to clarify `server responded with HTTP 200 but without any data.`

The server should always return the query object for 200 status. In the case of a incomplete with no results, it'll just be a query object with no buffers attached to the end.

Reply:
Ok, I see. I was planning to just return 0-data. In that case, the returned query object will be in the incomplete state and this logic should be re-worked.

Reply:
Done. This is no longer setting the 'INCOMPLETE' status. The expectation is that the server will return a query object in the incomplete state without any attached buffers. This path will not return any curl errors and will return directly to the user with an OK status.